lib/plugin/selenoid.js

```diff
@@ -0,0 +1,155 @@
+  const deleteVideoPromiseList = passedTests.map((test) => axios.delete(`${seleniumUrl}/video/${test}.mp4`));
+  const deleteLogPromiseList = passedTests.map((test) => axios.delete(`${seleniumUrl}/logs/${test}.log`));
+
+  return Promise.all(deleteVideoPromiseList.concat(deleteLogPromiseList)).then(() => output.debug('Deleted passed tests'));
+};
+
+const setSelenoidOptions = (config) => {
+  const WebDriver = container.helpers('WebDriver');
+  WebDriver._setConfig(Object.assign(WebDriver.options, {
+    capabilities: { 'selenoid:options': config },
+  }));
+};
+
+const replaceScriptConfig = (config) => {
+  for (const key of Object.keys(config)) {
+    dockerCreateScript = dockerCreateScript.replace(`$${key}$`, config[key]);
+  }
+  dockerStartScript = dockerStartScript.replace('$name$', config.name);
+  dockerStopScript = dockerStopScript.replace('$name$', config.name);
+};
+
+
+const selenoid = (config) => {
+  const {
+    autoStart, name = 'selenoid', deletePassed = true, additionalParams = '', autoCreate = true,
+  } = config;
+  const passedTests = [];
+  recorder.startUnlessRunning();
+  replaceScriptConfig({ name, additionalParams });
+
+  if (autoStart) {
+    event.dispatcher.on(event.all.before, () => {
+      recorder.add('Starting selenoid', async () => {
+        return createAndStart(autoCreate).then(() => output.debug('Selenoid started')).catch((err) => { throw err; });
+      });
+    });
+
+    event.dispatcher.on(event.all.after, () => {
+      recorder.add('Stopping selenoid', async () => {
+        return wait(10000).then(() => deletePassedTests(passedTests)).then(stopSelenoid).then(() => output.debug('Selenoid stopped'))
+          .catch((err) => { throw err; });
+      });
+    });
+  }
+
+  event.dispatcher.on(event.all.before, () => {
+    setSelenoidOptions(config);
+  });
+
+  event.dispatcher.on(event.test.before, (test) => {
+    const WebDriver = container.helpers('WebDriver');
+    const { options } = WebDriver;
+    recorder.add('setting selenoid capabilities', () => {
+      options.capabilities['selenoid:options'].name = test.title;
+      options.capabilities['selenoid:options'].videoName = `${test.title}.mp4`;
+      options.capabilities['selenoid:options'].logName = `${test.title}.log`;
+      WebDriver._setConfig(options);
+    });
+  });
+
+  if (deletePassed) {
+    event.dispatcher.on(event.test.passed, (test) => {
+      passedTests.push(test.title);
+    });
+  }
+};
+
+/**
+ * # Selenoid
+ *
+ * Selenoid plugin with for recording video recording and logs
+ *
+ * ## Prerequisite
+ *
+ *  - Docker
+ *
+ * ## How to ?
+ *
+ * ### 1. Add browsers.json
+ * Add browsers.json parallel to codecept conf location. [Refer here](https://aerokube.com/selenoid/latest/#_prepare_configuration) to know more about browsers.json
+ *
+ * **You can download the sample json from example.**
+ *
+ * ### 2. Add plugin configuration in codecept
+ *
+ *  Add plugin config to codecept conf. 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4798440)